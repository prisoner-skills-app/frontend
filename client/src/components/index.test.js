import React from 'react';
import { mount, shallow } from 'enzyme';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';

//Testing Comopnents
import { Header, CandidateCard, LogIn, NavBar } from './index';

beforeEach(() => {
    jest.resetModules();
});

//Attempting to mock context API
// const withContext = () => {
//     jest.mock();
// }

test('Header snapshot', () => {
    let Dummy = () => <h1>Dummy</h1>;

    let header = render(
        <BrowserRouter>
            <Header
                title={`Test Title \n with line break`}
                direction="column"
                align="center"
                backButton
                dropdown={<Dummy />}
            />
        </BrowserRouter>
    );
    expect(header).toMatchSnapshot();
});

test('Candidate snapshot', () => {
    let candidateCard = render(
        <CandidateCard
            key="asdfasdf"
            name="John Wick"
            description="The best hitman the world has ever seen"
            skills="killing, hitting, not dying"
        />
    );
    expect(candidateCard).toMatchSnapshot();
});

// test('NavBar Snapshot', () => {
//
//     //TODO: Needs to somehow mock the context API for testing
//     let navBar = render(<NavBar />);
//     expect(navBar).toMatchSnapshot();
// });
