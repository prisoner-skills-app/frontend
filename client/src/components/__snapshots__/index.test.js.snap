// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Candidate snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui card"
        style="height: 300px; max-width: 280px; margin-bottom: 20px; margin-right: 30px;"
      >
        <div
          class="content"
        >
          <div
            class="header"
          >
            John Wick
          </div>
          <div
            class="description"
          >
            The best hitman the world has ever seen
          </div>
          <h4>
            Skills
          </h4>
          <div
            class="ui label"
          >
            killing
          </div>
          <div
            class="ui label"
          >
             hitting
          </div>
          <div
            class="ui label"
          >
             not dying
          </div>
        </div>
        <div
          class="content"
          style="display: flex; justify-content: center; align-items: center;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui card"
      style="height: 300px; max-width: 280px; margin-bottom: 20px; margin-right: 30px;"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          John Wick
        </div>
        <div
          class="description"
        >
          The best hitman the world has ever seen
        </div>
        <h4>
          Skills
        </h4>
        <div
          class="ui label"
        >
          killing
        </div>
        <div
          class="ui label"
        >
           hitting
        </div>
        <div
          class="ui label"
        >
           not dying
        </div>
      </div>
      <div
        class="content"
        style="display: flex; justify-content: center; align-items: center;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-EHOje cuJJGT"
        direction="column"
      >
        <div
          class="sc-ifAKCX eXgxdJ"
        >
          <i
            aria-hidden="true"
            class="caret left icon"
            style="margin-right: 10px;"
          />
          <h1
            class="sc-bZQynM fIEYMI"
          >
            <div
              style="text-align: center;"
            >
              Test Title 
            </div>
            <div
              style="text-align: center;"
            >
               with line break
            </div>
          </h1>
        </div>
        
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-EHOje cuJJGT"
      direction="column"
    >
      <div
        class="sc-ifAKCX eXgxdJ"
      >
        <i
          aria-hidden="true"
          class="caret left icon"
          style="margin-right: 10px;"
        />
        <h1
          class="sc-bZQynM fIEYMI"
        >
          <div
            style="text-align: center;"
          >
            Test Title 
          </div>
          <div
            style="text-align: center;"
          >
             with line break
          </div>
        </h1>
      </div>
      
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
